core/modules/system/system.theme.inc

```diff
@@ -657,3 +657,65 @@ function _token_token_tree_format_row($token, array $token_info, $is_group = FAL
 
   return $row;
 }
+
+/**
+ * Prepares variables for Page Component block templates.
+ */
+function system_preprocess_page_components(&$variables) {
+  $settings = $variables['settings'];
+  $child_delta = $variables['child_delta'];
+  $variables += array(
+    'action_links' => NULL,
+    'tabs' => NULL,
+    'messages' => NULL,
+    'title' => NULL,
+  );
+
+  if ($child_delta == 'title' || $child_delta == 'title_combo') {
+    if ($settings['title_tag'] == 'none') {
+      $variables['title'] = backdrop_get_title();
+    }
+    else {
+      $output = '<' . $settings['title_tag'];
+      if (strlen($settings['title_class'])) {
+        $output .= ' class="' . $settings['title_class'] . '"';
+      }
```

Comment:
In other places where we have dynamic title tags, we output the tag itself in the .tpl.php file. For example `block-dynamic.tpl.php` contains the following:

```
<?php if ($title): ?>
  <<?php print $title_tag; ?><?php print backdrop_attributes($title_attributes); ?>><?php print $title; ?></<?php print $title_tag; ?>>
<?php endif;?>
```

But it doesn't support a "none" option (though I'm not sure if that is necessary, shouldn't the title always have a tag?)